scripts/make_video.py

```diff
@@ -0,0 +1,154 @@
+
+
+def init_arg_parser():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument('--checkpoint_id',
+                        default="stabilityai/stable-diffusion-2-1",
+                        help="checkpoint id on huggingface")
+    parser.add_argument('--prompts', nargs='+',
+                        help='sequence of prompts')
+    parser.add_argument('--seeds', type=int, nargs='+',
+                        help='seed for each prompt')
+    parser.add_argument('--num_interpolation_steps', type=int, nargs='+',
+                        help='number of steps between each image')
+    parser.add_argument('--output_dir', default="dreams",
+                        help='output directory')
+    parser.add_argument('--name',
+                        help='output sub-directory')
+    parser.add_argument('--fps', type=int, default=10,
+                        help='frames per second')
+    parser.add_argument('--guidance_scale', type=float, default=7.5,
+                        help='diffusion guidance scale')
+    parser.add_argument('--num_inference_steps', type=int, default=50,
+                        help='number of diffusion inference steps')
+    parser.add_argument('--height', type=int, default=512,
+                        help='output image height')
+    parser.add_argument('--width', type=int, default=512,
+                        help='output image width')
+    parser.add_argument('--upsample', action='store_true',
+                        help='upscale x4 using Real-ESRGAN')
+    parser.add_argument('--batch_size', type=int, default=1,
+                        help='batch size')
+    parser.add_argument('--audio_filepath',
+                        help='path to audio file')
+    parser.add_argument('--audio_offsets', type=int, nargs='+',
+                        help='audio offset for each prompt')
+    parser.add_argument('--negative_prompt',
+                        help='negative prompt (one for all images)')
+
+    parser.add_argument('--cfg',
+                        help='yaml config file (overwrites other options)')
+
+    return parser
+
+
+def parse_args(parser):
+    args = parser.parse_args()
+
+    # read config file
+    if args.cfg is not None:
+        with open(args.cfg) as f:
+            cfg = yaml.safe_load(f)
+        for key, val in cfg.items():
+            if hasattr(args, key):
+                setattr(args, key, val)
+            else:
+                raise ValueError(f'bad field in config file: {key}')
+
+    # check for prompts
+    if args.prompts is None:
+        raise ValueError('no prompt provided')
+    if args.seeds is None:
+        args.seeds = [random.getrandbits(16) for _ in args.prompts]
+
+    # check audio arguments
+    if args.audio_filepath is not None and args.audio_offsets is None:
+        raise ValueError('must provide audio_offsets when providing '
```

Comment:
makes me wonder if this should just be raised in the pipeline code itself instead of the parser (if its not already)

Reply:
same goes for many of the other raised errors in this script

Reply:
Maybe. That's a design question IMO. Do we want to raise errors to the unadvised CLI user as early as possible, while trusting that the developer who writes his owns scripts knows what they are doing? Or do we want to raise errors as close to the problematic code/as late as possible but such that it propagates?

Reply:
Agreed. I'm fine with the way you did it here :)

Reply:
Reason I say it though is that `walk` used to be a CLI interface when I first made this repo, so it should be the fn catching all the cases...but we can do it this way for now instead, I'm not picky.